src/query/functions/binary/binary.go

```diff
@@ -0,0 +1,234 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package binary
+
+import (
+	"github.com/m3db/m3/src/query/block"
+	"github.com/m3db/m3/src/query/executor/transform"
+)
+
+type binaryFunc func(x, y float64) float64
+type singleScalarFunc func(x float64) float64
+
+// processes two logical blocks, performing a logical operation on them
+func processBinary(
+	lhs, rhs block.Block,
+	params NodeParams,
+	controller *transform.Controller,
+	fn binaryFunc,
+) (block.Block, error) {
+	lIter, err := lhs.StepIter()
+	if err != nil {
+		return nil, err
+	}
+
+	// fn := n.op.fn
+	// params := n.op.params
```

Comment:
Remove?

Reply:
Oops, missed this. Will do it in a future commit